openid-4-verifiable-presentations-1_0.md

```diff
@@ -542,11 +548,250 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
+no specific restrictions are placed on the metadata or validity of the requested
+Credential.
+
+`expect_claims`:
+: OPTIONAL. An array that specifies the claims that the requested
+Verifiable Credential must contain. Each entry in the array corresponds to exactly
+one claim that is to be returned by the Wallet to fulfill the credential
+request. 
+
+`alternative_credentials`:
+: OPTIONAL. An array of credential group identifiers that specify alternative
+groups of credentials that MAY be returned instead of the credential. Any credential
+in a group listed in `alternative_credentials` MUST NOT itself contain an
+`alternative_credentials` property.
+
+Each entry MUST be an object with the following properties:
+
+`group`:
+: OPTIONAL. A string identifying a group of claims for expressing alternatives
+between claims. Multiple claims MAY have the same group identifier. The group
+identifier for claims is not reflected in the response.
+
+`path`:
+: REQUIRED if the credential format uses a JSON-based claims structure; MUST NOT
+be present otherwise. The value MUST be a claims path pointer that specifies the path to the claim
+within the Verifiable Credential, as defined in (#claims_path_pointer).
+
+`namespace`:
+: REQUIRED if the credential format is based on ISO 18013-5; MUST NOT be present otherwise. 
+The value MUST be a string that specifies the namespace of the claim
+within the Verifiable Credential, e.g., `org.iso.18013.5.1`.
+
+`claim_name`:
+: REQUIRED if the credential format is based on ISO 18013-5; MUST NOT be present otherwise. 
+The value MUST be a string that specifies the name of the claim within the provided namespace
+in the Verifiable Credential, e.g., `first_name`.
+
+`value`:
+: OPTIONAL. A string, integer or boolean value that specifies the expected value of the claim. If the
+`value` property is present, the Wallet MUST return the claim only if the type and value
+of the claim match the type and value specified in the query.
+
+`values`:
+: OPTIONAL. An array of strings, integers or boolean values that specifies the expected values of the claim.
+If the `values` property is present, the Wallet MUST return the claim only if the
+type and value of the claim both match for at least one of the elements in the array.
+
+`alternative_claims`:
+: OPTIONAL. An array of claim group identifiers that specify alternative groups of claims
+that MAY be returned instead of the claim specified by the `path` or `claim_name` property if
+that claim is not present or does not match the restriction expressed by `value` or `values`. Any
+claim in a group listed in `alternatives` MUST NOT itself contain an `alternatives` property.
+
+### Semantics of `alternative_claims` and `alternative_credentials`
+
+By default, each element within `expect_credentials` or `expect_claims` MUST be returned, unless:
+
+- The element contains an `alternative_claims` or `alternative_credentials`
```

Comment:
To me it feels a bit weird that alternative_claims is added to an expected claim. As this is a two way relation (you could also add it to the other claim). 

I think separating the requirements from the credentials is one of the things that really make sense to me in PEX, as it gives a nice structure:
- define which credentials can be presented.
- define which combination of credentials are allowed.

I feel this syntax intertwines it more and loses readability over it, while also making it quite hard to spot invalid configurations (as a claim/cred referenced as alternative can't reference alternatives itself). Would be interesting to see the complexity of implementing such an approach (hard to wrap my head around at the moment)